### DISCUSSION TITLE
skip checkup Xcode version: error executing xcodebuild

### DISCUSSION THREAD

**Discussion by hamza5645** [skiptools/skip#67] [Q&A]
<img width="842" alt="Screenshot 2023-12-17 at 2 19 10 PM" src="https://github.com/skiptools/skip/assets/97033430/54e7ae2b-9c06-46e6-b3ba-6491d4f37c57">

Xcode version: error executing xcodebuild
Test Swift (Failed)
Test Kotlin (Failed)
Archive iOS ipa (Failed)

**Comment by marcprux** [comment]


**Reply by hamza5645** [reply]
Command line tools are already installed.

**Reply by hamza5645** [reply]
<img width="1470" alt="Screenshot 2023-12-17 at 2 31 13 PM" src="https://github.com/skiptools/skip/assets/97033430/350537e5-b9c8-4889-b6bc-d9fd0a6d6c09">

These are the errors I got after running skip checkup -v

**Reply by marcprux** [reply]
That's an odd error. Is Xcode 15 itself installed?

**Reply by hamza5645** [reply]
yes, Xcode 15.1 is installed from the app store and it works perfectly fine

**Reply by hamza5645** [reply]
I tried brew reinstalling skip but got the same errors

**Reply by marcprux** [reply]
If you manually run `swift package init` in a newly created directory (like "swifttest") and then run `swift test`, does it fail with the same error?

**Reply by hamza5645** [reply]
<img width="842" alt="Screenshot 2023-12-17 at 2 55 20 PM" src="https://github.com/skiptools/skip/assets/97033430/a4e03a90-49f9-4ed3-bf8f-1d360a335405">

<img width="1032" alt="Screenshot 2023-12-17 at 2 55 40 PM" src="https://github.com/skiptools/skip/assets/97033430/7c9793ca-3ad6-4c04-823f-9d6496d04454">

**Reply by marcprux** [reply]
Might you also have an older version of Xcode installed alongside Xcode 15.1? What does `xcrun --show-sdk-path --sdk macosx` show you? If it shows an older version of Xcode, you can switch to the new one with: `sudo xcode-select -switch /Applications/Xcode.app/Contents/Developer` and then try the `swift test` command again.

**Reply by hamza5645** [reply]
This worked the swift test didn't throw any errors and the skip checkup worked.
Thanks a lot for your effort.

**Reply by marcprux** [reply]
Fabulous!
